### PR TITLE
THU-108: Tools should have timeouts

### PR DESCRIPTION
**Pull request by raivieiraadriano92**
> [!NOTE]
> Adds 5s timeouts to all Thunderbolt Pro tool requests and introduces comprehensive unit tests covering success, failure, and edge cases.
> 
> - **Thunderbolt Pro Tools**:
>   - Add `timeout: 5000` to all `ky.post` calls in `src/integrations/thunderbolt-pro/tools.ts` (`search`, `fetchContent`, `getCurrentWeather`, `getWeatherForecast`, `searchLocations`).
> - **Tests**:
>   - New `src/integrations/thunderbolt-pro/tools.test.ts` with unit tests for `search`, `fetchContent`, `getCurrentWeather`, `getWeatherForecast`, and `searchLocations`:
>     - Success paths, default params, API failures, network/timeout errors, malformed/empty/large responses, and invalid forecast data validation.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7f7a793a1b86fedbdb74d91a4869efcc37a4f5c4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/integrations/thunderbolt-pro/tools.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import type { ToolConfig } from '@/types'
 import ky from 'ky'
 import { z } from 'zod'
 
+const REQUEST_TIMEOUT = 15000
```
Comment (cjroth) [inline src/integrations/thunderbolt-pro/tools.ts]: I'd go with camel case here just to be consistent with the rest of the codebase (I know the all caps convention for constants, but I think this is just a local constant for now)

Comment (raivieiraadriano92) [reply src/integrations/thunderbolt-pro/tools.ts]: done

File: src/integrations/thunderbolt-pro/tools.test.ts [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+        success: true,
+      }
+
+      mockJson.mockResolvedValue(mockResponse)
+
+      const result = await getWeatherForecast(params)
+
+      expect(result).toEqual(mockWeatherForecastData)
+      expect(mockPost).toHaveBeenCalledWith(
+        'https://api.thunderbolt.com/pro/weather/forecast',
```
Comment (cjroth) [inline src/integrations/thunderbolt-pro/tools.test.ts]: this might be nitpicky, but i think we should do something like example.com because we don't own this domain and I don't want AI to see this and start hardwiring a domain that we don't own into the code haha

Comment (raivieiraadriano92) [reply src/integrations/thunderbolt-pro/tools.test.ts]: good one haha
it's done

**Review by cjroth — APPROVED** [review]
